src/wrappers/gcc_wrapper.cpp

```diff
@@ -290,6 +297,9 @@ std::map<std::string, expected_file_t> gcc_wrapper_t::get_build_files() {
   if (has_coverage_output(m_resolved_args)) {
     files["coverage"] = {file::change_extension(files["object"].path(), ".gcno"), true};
   }
+  if (generate_makefile_dependencies) {
+    files["d"] = {file::change_extension(files["object"].path(), ".d"), true};
```

Comment:
I think this patch is wrong and should not be applied.

The dependencies often contain absolute paths. Thus caching dependency files might be harmful.
Suppose user A builds project in `/home/A/proj`. After that user B tries to build the same project in `/home/B/proj`, and gets dependencies wrong.

On the other hand possible gains look pretty minimal. Any source should be preprocessed to compute the hash. When preprocessing the compiler opens every `#include` file anyway. Recording every included file and printing out that information as a Makefile fragment doesn't look like a resource intensive task.

Reply:
Good point about the absolute paths! I will have to go through this once more. I still don't like the fact that `.d`-files are currently generated as a side effect of the pre-processor step. It's one of those things that could easily change in the future (e.g. if we implement a "direct mode" without involving the pre-processor).

Reply:
Firstly in a direct mode buildcache would have to figure out included files itself, and could generate the dependency files from that info.

Secondly I'm not sure if we should bother with a "direct mode".  I haven't observed any measurable performance gains with C++ projects (and direct mode). And this is sort of an expected result for two reasons:
1. Compiling a single C++ source (especially with templates) is more resource intensive than preprocessing the whole project.
2. Those object files need to be linked, which is more resource intensive than preprocessing the whole project.

I haven't noticed any substantial gains with C only projects (Linux kernel, samba) either.